presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -472,14 +473,14 @@ public boolean isDone()
     public boolean transitionToPlanning()
     {
         queuedTime.compareAndSet(null, nanosSince(createNanos).convertToMostSuccinctTimeUnit());
+        totalPlanningStartNanos.compareAndSet(null, System.nanoTime());
         return queryState.compareAndSet(QUEUED, PLANNING);
     }
 
     public boolean transitionToStarting()
     {
-        Duration durationSinceCreation = nanosSince(createNanos).convertToMostSuccinctTimeUnit();
-        queuedTime.compareAndSet(null, durationSinceCreation);
-        totalPlanningTime.compareAndSet(null, durationSinceCreation);
+        queuedTime.compareAndSet(null, nanosSince(createNanos).convertToMostSuccinctTimeUnit());
+        recordPlanningTime();
```

Comment:
I can see pros and cons for inlining vs extracting this method. I think inlining is a better idea given the simplicity of the method and the repetitive nature of other code around this line.